marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java

```diff
@@ -517,18 +524,21 @@ private void updateContentsFromMavenXML(Product product, String metadataContent,
 
     updateProductCompatibility(product, mavenVersions);
 
-    List<String> currentVersions = product.getReleasedVersions();
-    if (CollectionUtils.isEmpty(currentVersions)) {
-      product.setReleasedVersions(new ArrayList<>());
-      currentVersions = productModuleContentRepo.findVersionsByProductId(product.getId());
-    }
-    mavenVersions = mavenVersions.stream().filter(filterNonPersistVersion(currentVersions)).toList();
+    List<String> currentVersions = ObjectUtils.isNotEmpty(product.getReleasedVersions()) ?
+        product.getReleasedVersions() :
+        productModuleContentRepo.findVersionsByProductId(product.getId());
+
+    mavenVersions = mavenVersions.stream().filter(version -> !currentVersions.contains(version)).toList();
 
     List<ProductModuleContent> productModuleContents = new ArrayList<>();
     for (String version : mavenVersions) {
-      product.getReleasedVersions().add(version);
-      handleProductArtifact(version, product.getId(), productModuleContents, mavenArtifact,
+      Optional.ofNullable(product.getReleasedVersions()).orElseGet(() -> {
+            product.setReleasedVersions(new ArrayList<>());
```

Comment:
consider to move this set outside, it not make sense if you put it inside for loop